### PR TITLE
Delete event comment (#3209)

### PR DESCRIPTION
**Pull request by OlegVat**
Oleh Vatuliak
issue #3209

## Summary of issue

Added request to delete user's event comment by id

## Summary of change

The user now can delete their event comments

### CODE REVIEW

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 1]
```diff
@@ -91,4 +93,26 @@ public PageableDto<EventCommentDto> getAllActiveComments(Pageable pageable, User
...
+    /**
+     * Method to delete comment {@link EventComment} by Id.
+     *
+     * @param eventCommentId specifies {@link EventComment} to which we search for
+     *                       comments.
+     */
+    @Override
+    public void delete(Long eventCommentId, UserVO user) {
+        Optional<EventComment> eventComment = eventCommentRepo.findById(eventCommentId);
+
```
Comment (danylohlynsky) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java:106]:
```suggestion
EventComment eventComment = eventCommentRepo.findById(eventCommentId).orElseThrow(() -> new NotFoundException(message))
```

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 2]
```diff
@@ -91,4 +93,26 @@ public PageableDto<EventCommentDto> getAllActiveComments(Pageable pageable, User
...
+    /**
+     * Method to delete comment {@link EventComment} by Id.
+     *
+     * @param eventCommentId specifies {@link EventComment} to which we search for
+     *                       comments.
+     */
+    @Override
+    public void delete(Long eventCommentId, UserVO user) {
+        Optional<EventComment> eventComment = eventCommentRepo.findById(eventCommentId);
```
Comment (danylohlynsky) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java]:
```suggestion
EventComment eventComment = eventCommentRepo.findById(eventCommentId).orElseThrow(() -> new NotFoundException(message))
```

Comment (OlegVat) [reply service/src/main/java/greencity/service/EventCommentServiceImpl.java]: Done

**Review by danylohlynsky — COMMENTED** [review]
Please increase code coverage

**Review by danylohlynsky — APPROVED** [review]

File: core/src/main/java/greencity/controller/EventCommentController.java [hunk 3]
```diff
@@ -17,13 +17,7 @@
...
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/EventCommentController.java]: Please do not use wildcard

Comment (OlegVat) [reply core/src/main/java/greencity/controller/EventCommentController.java]: Done

File: service/src/test/java/greencity/service/EventCommentServiceImplTest.java [hunk 4]
```diff
@@ -119,4 +118,47 @@ void getAllActiveCommentsEventNotFoundException() {
...
+
+    @Test
+    void delete() {
+        UserVO userVO = getUserVO();
+        Long commentId = 1L;
+
+        when(eventCommentRepo.findById(commentId))
+            .thenReturn(Optional.ofNullable(ModelUtils.getEventComment()));
+
+        eventCommentService.delete(commentId, userVO);
```
Comment (danylohlynsky) [inline service/src/test/java/greencity/service/EventCommentServiceImplTest.java:130]: Please add some verify to remove code smell

**Review by innayashna — APPROVED** [review]

File: service/src/test/java/greencity/service/EventCommentServiceImplTest.java (hunk 4 above)
Comment (OlegVat) [reply service/src/test/java/greencity/service/EventCommentServiceImplTest.java:130]: Done

**Review by AndriyUrban — APPROVED** [review]

**Pull request merged**